### PR TITLE
feat: Add and view exercises

### PR DESCRIPTION
**Pull request by mcgarj**
# Add and View exercises

This PR adds core functionality to let users add and delete exercises as per:
 - [view exercises](https://trello.com/c/jD6YSv6R)
 - [delete exercises](https://trello.com/c/V5dFZtDx)
 - [add exercises](https://trello.com/c/p9z6wpbj)

These were meant to be separate PRs but upon looking into SwiftData, the functionality to add and delete were very similar, so for UI testing purposes it all became 1 PR

# Checklist

## Before raising your pull request:
- [ ] Update the documentation to reflect your changes
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with ticket ID and a short description about the feature or update
      i.e. _TRELLO_TICKET: Added save functionality for exercises_

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Trello
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- ~[ ] Met all accessibility requirements?~
    - [ ] Checked dynamic type sizes are applied
    - [ ] Checked VoiceOver can navigate your new code
    - [ ] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [x] Ran the app to ensure that no regressions have been caused by changes during code review.
- [x] Targeted the correct branch; `develop`, `release` or `main`

### CODE REVIEW

File: ExerciseTracker/Tests/UITests/ExerciseTrackerUITestsLaunchTests.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import XCTest
```
Comment (beccapearce) [inline ExerciseTracker/Tests/UITests/ExerciseTrackerUITestsLaunchTests.swift]: Just wondering if this file is needed, it looks like just an import and a commented out test (but I could be missing some swift magic!)

**Comment by beccapearce** [comment]
It might be worth looking at rebasing so commits such as #426535ddaf69965d828f125c371070d1f6b5ca79 and #8286d7fca9fc5122172c8448c994f8a234443cb0 can be merged, (not for this PR since it's merged but it's a usful thing to learn.

In this case you would run `git rebase -i main develop` which will give you a list of commits that you have made in comparison to your develop branch, you can then choose to drop #426535ddaf69965d828f125c371070d1f6b5ca79 by replacing the `p` next to the commit SHA with a `d` (p stands for pick and will include the changes and d stands for drop and will get rid of all those changes)

**Comment by beccapearce** [comment]
This is all looking good to me! 🚀

File: ExerciseTracker/Tests/UITests/ExerciseTrackerUITestsLaunchTests.swift (hunk 1 above)
Comment (mcgarj) [reply ExerciseTracker/Tests/UITests/ExerciseTrackerUITestsLaunchTests.swift]: You're spot on, that one isn't needed in my case (from my reading this file is used more for snapshot testing). I think it should have been deleted before the big merge happened. The commenting out was me initially testing if it broke anything by not being there as I hadn't seen it before 😅

**Pull request merged**